### DISCUSSION TITLE
Anyone willing to ELI5 the Selenium web driver please?

### DISCUSSION THREAD

**Discussion by Alllfff** [robbrad/UKBinCollectionData#461] [Q&A] [answered]
My council requires "URL of the remote Selenium web driver to use" but I have no idea how to do this even after checking all the links provided, it is just beyond my computer skill level. Anyone care to write a step by step or point me in some guide or video aimed at complete novices please?
Thank you

**Comment by OliverCullimore** [comment]
@Alllfff it can be installed in various ways so to point you to the right guide it would be useful to know the following:

Do you have Home Assistant installed? If so, which install method from [this list](https://www.home-assistant.io/installation/) did you use?

Do you have any other hardware such as a Raspberry Pi you can install things on?

**Reply by Alllfff** [reply]
Hi
I do have Home Assistant installed, as per DIY with Raspberry Pi from the link you provided. 
I have a spare RPi model 2 or maybe even 3, I would need to check.
I also have iMac and a mini Mac both running 24/7 on the network.

**Comment by mrw298** [comment]
@Alllfff I found it a bit of pain also, so I knocked up a custom local addon so it could run on my HAOS instance.

In case it's useful to anyone else:
- Create a folder selenium_addon under addons
- Drop these files in, stripping the .txt file off the end
-- [stop_trap.sh.txt](https://github.com/robbrad/UKBinCollectionData/files/13565033/stop_trap.sh.txt)
-- [entrypoint.sh.txt](https://github.com/robbrad/UKBinCollectionData/files/13565035/entrypoint.sh.txt)
-- [Dockerfile.txt](https://github.com/robbrad/UKBinCollectionData/files/13565036/Dockerfile.txt)
-- [config.yaml.txt](https://github.com/robbrad/UKBinCollectionData/files/13565037/config.yaml.txt)
- Refresh your addon repos

You should then have a local addon of selenium you can point the integration at (http://local-selenium-addon:4444/wd/hub)

**Reply by nraimo1** [reply, marked as answer]
I've tried this won't work,  can't seem to find it as a local addon then tried turning it into a github file to load addon that way it wouldn't load https://github.com/nraimo1/selenium_addon  any suggestions

**Reply by mrw298** [reply]
sorry, I followed the instructions here: https://developers.home-assistant.io/docs/add-ons/tutorial substituting my files and it worked for me.

**Reply by nraimo1** [reply]
whats the LOCATIOn of the addons file could you screenshot it?  Mine didn't exist anywhere so I assume I create one in a directory somewhere?

**Reply by nraimo1** [reply]
I have it here "homeassistant/addons/selenium_addon"

**Reply by mrw298** [reply]
think you need it up a level
![image](https://github.com/robbrad/UKBinCollectionData/assets/3275620/b5b15e0c-43c4-4949-bd5c-5d08eeb53b89)

**Reply by nraimo1** [reply]
weird!  add it still doesn't show,  reboot system and folder deletes it self!

**Reply by Alllfff** [reply]
still not working for me :(

how do you refresh addon repos?

any way to check if the addon is in place and functioning?

The supervisor is getting "invalid build environment" error

**Reply by Alllfff** [reply]
IT WORKS :) thank you!!! it takes a long time to install and it seems it is not doing anything. also had to restart home assistant

**Reply by rajp1503** [reply]
Is the path where I need to put the files

homeassistant/addons/selenium_addon/

I believe it needs to go 1 level up.. how do I move the files to that??

**Reply by rajp1503** [reply]
OK so I was able to change the setting in File Editor to expose the root folder. Moved the addons/selenium_addon folder there and the contents of the file.. I still do not get anything happening when I try http://local-selenium-addon:4444/wd/hub. How do I see if this addon has worked or not?

**Reply by Alllfff** [reply]
Go to your addons in HA and see if it is there...you may need to run it or install it and run it from there...

**Reply by rajp1503** [reply]
> Go to your addons in HA and see if it is there...you may need to run it or install it and run it from there...

I don't see it in AddOns. I also noticed that the addon folder that I created in the root has also disappeared.
